frontend/document/documentregistry.lua

```diff
@@ -9,22 +9,47 @@ local DocumentRegistry = {
     providers = {},
 }
 
-function DocumentRegistry:addProvider(extension, mimetype, provider)
-    table.insert(self.providers, { extension = extension, mimetype = mimetype, provider = provider })
+function DocumentRegistry:addProvider(extension, mimetype, provider, weight)
+    table.insert(self.providers, {
+        extension = extension,
+        mimetype = mimetype,
+        provider = provider,
+        weight = weight or 100,
+    })
 end
 
---- Returns the registered document handler.
+--- Returns the preferred registered document handler.
 -- @string file
 -- @treturn string provider, or nil
 function DocumentRegistry:getProvider(file)
+    local providers = self:getProviders(file)
+
+    if #providers >= 1 then
```

Comment:
This test should probably be in getProviders(), just return nil instead of an empty table.